src/solvers/lowering/byte_operators.cpp

```diff
@@ -100,13 +185,109 @@ static exprt unpack_rec(
         throw "cannot unpack struct with non-byte aligned components:\n"+
           struct_type.pretty();
 
+      exprt new_offset = nil_exprt();
+      mp_integer new_offset_int;
+      exprt new_max_bytes = nil_exprt();
+      mp_integer new_max_bytes_int;
+
+      if(!to_integer(offset, new_offset_int))
+      {
+        new_offset_int -= member_offset_bits % 8;
+        if(new_offset_int >= 0)
+          new_offset = from_integer(new_offset_int, offset.type());
+      }
+
+      if(!to_integer(max_bytes, new_max_bytes_int))
```

Comment:
Comment before these what the intended adjustment is? At the moment it looks like you would reduce *both* lower and upper bounds if the access is not byte-aligned, whereas I might expect you to round the lower bound down and the upper bound up to ensure you include the desired range?

Reply:
No, byte-alignedness is tested above, there shouldn't be a need for rounding I believe.

Reply:
In that case `member_offset_bits % 8` is always zero, as it starts at zero and is the incremental sum of several `element_width`s, each of which is itself a multiple of eight.

Reply:
Your comment really highlighted that there was a problem: `member_offset_bits` wasn't actually being used properly, this should have been a division, not modulo!